Civi/Core/SettingsBag.php

```diff
@@ -168,6 +168,7 @@ public function all() {
         [$this->defaults, $this->values, $this->mandatory]
       );
     }
+    $this->combined['contribution_invoice_settings'] = $this->getContributionSettings();
```

Comment:
Would this work if it is was inside of the ` if ($this->combined === NULL) {` block?

The reason I ask.. `$this->combined` is supposed to be a cache. The `get($key)` function calls `all()` under the expectation that `all()` can usually be executed very quickly. Adding `getContributionSettings()` at this spot would mean that it's recreating a 10-element array whenever someone does a `get($key)`.

Reply:
@totten maybe - but this should be temporary